### PR TITLE
It would be handy to reuse HttpClient class functionality for not onl…

### PR DESCRIPTION
**Pull request by rvgulinski**
…y V2 API endpoints. HttpClient can be extended now to provide different authentication strategies.

### CODE REVIEW

File: src/main/java/com/getbase/http/HttpClient.java [hunk 1]
```diff
@@ -86,6 +86,10 @@ public Response request(HttpMethod method,
         return response;
     }
 
+    protected Request.Builder authorizationStrategy(Request.Builder requestBuilder) {
```
Comment (michal-mally) [inline src/main/java/com/getbase/http/HttpClient.java]: Maybe we could call it `applyAuthorization()`?

File: src/test/groovy/com/getbase/http/CustomAuthorizationHttpClientTest.groovy [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package com.getbase.http
+
+import com.getbase.Configuration
+import spock.lang.Specification
+
+class CustomAuthorizationHttpClientTest extends Specification {
```
Comment (michal-mally) [inline src/test/groovy/com/getbase/http/CustomAuthorizationHttpClientTest.groovy:6]: By convention it shall end w/ `Spec`.

Comment (rvgulinski) [inline src/test/groovy/com/getbase/http/CustomAuthorizationHttpClientTest.groovy:6]: It's true but I didn't want to break existing Test convention.

File: src/main/java/com/getbase/http/HttpClient.java (hunk 1 above)
Comment (rvgulinski) [inline src/main/java/com/getbase/http/HttpClient.java]: 👍

File: src/main/java/com/getbase/http/HttpClient.java [hunk 3]
```diff
@@ -86,6 +86,10 @@ public Response request(HttpMethod method,
         return response;
     }
 
+    protected Request.Builder applyAuthorization(Request.Builder requestBuilder) {
```
Comment (pawelgocek) [inline src/main/java/com/getbase/http/HttpClient.java:90]: As we don't mention this feature in README we could at least add a comment here, like "Override this method if you want to use custom authorization method".

**Pull request merged**